runtime/signals_nat.c

```diff
@@ -58,43 +57,69 @@ extern void caml_win32_overflow_detection();
    Only generated assembly code can call [caml_garbage_collection],
    via the caml_call_gc assembly stubs.  */
 
-void caml_garbage_collection(void)
+void caml_garbage_collection()
```

Comment:
Please leave the explicit `(void)`!  With `()` this becomes a K&R function that can be called with arbitrarily many arguments.  Stick to ISO C, please.
```suggestion
void caml_garbage_collection(void)
```

Reply:
I've done an audit of the runtime for functions with the prefix `caml_`, and have introduced explicit `(void)` when necessary in https://github.com/ocaml/ocaml/pull/10831/commits/6aa1379ba5a4891e4eb76ad6baafcb02ecd616a8. There is likely to be a few more functions in the runtime that need to be fixed.